### PR TITLE
Feat (Cognito) support Lambda Trigger

### PR DESCRIPTION
**Pull request by ronnathaniel**
Screen Shot 2021-05-03 at 6 34 09 PM

### CODE REVIEW

File: epsagon/triggers/aws_lambda.py [hunk 1]
```diff
@@ -453,6 +453,60 @@ def __init__(self, start_time, event, context):
         }
 
 
+class CognitoLambdaTrigger(BaseLambdaTrigger):
+    """
+    Represents a Cognito Lambda Trigger
+    """
+    RESOURCE_TYPE = 'cognito'
+
+    def __init__(self, start_time, event, context):
```
Comment (ranrib) [inline epsagon/triggers/aws_lambda.py:462]:
```suggestion
    def __init__(self, start_time, event, _context):
```

File: epsagon/triggers/aws_lambda.py [hunk 2]
```diff
@@ -453,6 +453,60 @@ def __init__(self, start_time, event, context):
...
+        """
+
+        super(CognitoLambdaTrigger, self).__init__(start_time)
+
+        self.event_id = 'cognito-{uid}'.format(uid=str(uuid4))
+        self.resource['name'] = event.get('userPoolId')
+        self.resource['operation'] = event.get('triggerSource')
+
+        self.resource['metadata'] = {
+            'region': event.get('region'),
```
Comment (ranrib) [inline epsagon/triggers/aws_lambda.py:477]:
```suggestion
            'region': event.get('aws.region'),
```

File: epsagon/triggers/aws_lambda.py [hunk 3]
```diff
@@ -453,6 +453,60 @@ def __init__(self, start_time, event, context):
...
+            'username': event.get('userName'),
```
Comment (ranrib) [inline epsagon/triggers/aws_lambda.py:478]:
```suggestion
            'username': event.get('aws.cognito.userName'),
```

File: epsagon/triggers/aws_lambda.py [hunk 4]
```diff
@@ -453,6 +453,60 @@ def __init__(self, start_time, event, context):
...
+            'client_id': event.get('callerContext', {}).get('clientId'),
```
Comment (ranrib) [inline epsagon/triggers/aws_lambda.py:479]:
```suggestion
            'aws.cognito.client_id': event.get('callerContext', {}).get('clientId'),
```

File: epsagon/triggers/aws_lambda.py [hunk 5]
```diff
@@ -453,6 +453,60 @@ def __init__(self, start_time, event, context):
...
+            'user_pool_id': event.get('userPoolId'),
```
Comment (ranrib) [inline epsagon/triggers/aws_lambda.py:480]:
```suggestion
            'aws.cognito.user_pool_id': event.get('userPoolId'),
```

File: epsagon/triggers/aws_lambda.py [hunk 6]
```diff
@@ -453,6 +454,60 @@ def __init__(self, start_time, event, context):
...
+        """
+        Initialize Trigger
+        :param start_time: event's start time (epoch unix)
+        :param event: event dict from entry point
+        :param context: context dict from entry point
+        """
+
+        super(CognitoLambdaTrigger, self).__init__(start_time)
+
+        self.event_id = 'cognito-{uid}'.format(uid=str(uuid4))
```
Comment (ophiryael) [inline epsagon/triggers/aws_lambda.py:473]: Shouldn't you invoke the uuid4 function?

Comment (ronnathaniel) [reply epsagon/triggers/aws_lambda.py:473]: wow good catch. indeed

**Review by ophiryael — APPROVED** [review]

**Comment by ronnathaniel** [comment]
Travis tests pass for 2.7 & 3.7
3.6 & 3.8 failed because of too many buckets
`An error occurred: ServerlessDeploymentBucket - You have attempted to create more buckets than allowed (Service: Amazon S3; Status Code: 400; Error Code: TooManyBuckets; Request ID: SQQ9329VZRN65PTH; S3 Extended Request ID: Gx9uATTWPiqP4ZgvabgsqVjkwOpelns6/kwHLKm19pR9AQhfsNbwpHPiMG4EuyOwhZ5AyPltE+g=; Proxy: null).`

**Comment by ophiryael** [comment]
> Travis tests pass for 2.7 & 3.7
> 3.6 & 3.8 failed because of too many buckets
> `An error occurred: ServerlessDeploymentBucket - You have attempted to create more buckets than allowed (Service: Amazon S3; Status Code: 400; Error Code: TooManyBuckets; Request ID: SQQ9329VZRN65PTH; S3 Extended Request ID: Gx9uATTWPiqP4ZgvabgsqVjkwOpelns6/kwHLKm19pR9AQhfsNbwpHPiMG4EuyOwhZ5AyPltE+g=; Proxy: null).`
> 
> @ophiryael good to merge?

Restarted the job after some buckets were deleted.
You are good to go.

**Pull request merged**